### ISSUE TITLE
[shell] Quoted values break zsh regexp conditional expressions

### ISSUE THREAD

**Issue by disco0** [jeff-hykin/better-shell-syntax#9]
Unfortunately the terminating quote begins a quoted capture after the regex, and breaks highlighting until the next matching quotation character. 

An example in mpv's completion script `_mpv`:

<img width="834" alt="Screen Shot 2020-12-04 at 2 32 39 PM" src="https://user-images.githubusercontent.com/13393582/101222620-fb7bd100-363e-11eb-9b79-45ef66751653.png">

As well as a smaller test case, with issue case on bottom (`-regex-match` is not captured with any grammar—mostly using it to force capturing as quoted string)

``` zsh
[[ "text" -regex-match '^a([^a]+)a([^a]+)a' ]];
[[ "text" =~           '^a([^a]+)a([^a]+)a' ]];

[[ $var -regex-match '^a([^a]+)a([^a]+)a' ]];
[[ $var =~           '^a([^a]+)a([^a]+)a' ]];

[[ $desc -regex-match '^alias for --(\S+)' ]];
[[ $desc =~           '^alias for --(\S+)' ]];

```

<details>
<summary> Captured syntax tokens </summary> 

``` json
[
    {
        "c": "[[",
        "t": "source.shell source.shell meta.scope.logical-expression.shell punctuation.definition.logical-expression.shell"
    },
    {
        "c": " ",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "\"",
        "t": "source.shell source.shell meta.scope.logical-expression.shell string.quoted.double.shell punctuation.definition.string.begin.shell"
    },
    {
        "c": "text",
        "t": "source.shell source.shell meta.scope.logical-expression.shell string.quoted.double.shell"
    },
    {
        "c": "\"",
        "t": "source.shell source.shell meta.scope.logical-expression.shell string.quoted.double.shell punctuation.definition.string.end.shell"
    },
    {
        "c": " ",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "-r",
        "t": "source.shell source.shell meta.scope.logical-expression.shell keyword.operator.logical.shell"
    },
    {
        "c": "egex-match ",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "'",
        "t": "source.shell source.shell meta.scope.logical-expression.shell string.quoted.single.shell punctuation.definition.string.begin.shell"
    },
    {
        "c": "^a([^a]+)a([^a]+)a",
        "t": "source.shell source.shell meta.scope.logical-expression.shell string.quoted.single.shell"
    },
    {
        "c": "'",
        "t": "source.shell source.shell meta.scope.logical-expression.shell string.quoted.single.shell punctuation.definition.string.end.shell"
    },
    {
        "c": " ",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "]]",
        "t": "source.shell source.shell meta.scope.logical-expression.shell punctuation.definition.logical-expression.shell"
    },
    {
        "c": ";",
        "t": "source.shell source.shell punctuation.terminator.statement.semicolon.shell"
    },
    {
        "c": "[[",
        "t": "source.shell source.shell meta.scope.logical-expression.shell punctuation.definition.logical-expression.shell"
    },
    {
        "c": " ",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "\"",
        "t": "source.shell source.shell meta.scope.logical-expression.shell string.quoted.double.shell punctuation.definition.string.begin.shell"
    },
    {
        "c": "text",
        "t": "source.shell source.shell meta.scope.logical-expression.shell string.quoted.double.shell"
    },
    {
        "c": "\"",
        "t": "source.shell source.shell meta.scope.logical-expression.shell string.quoted.double.shell punctuation.definition.string.end.shell"
    },
    {
        "c": " ",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "=~",
        "t": "source.shell source.shell meta.scope.logical-expression.shell keyword.operator.logical.shell"
    },
    {
        "c": "           '",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "^",
        "t": "source.shell source.shell meta.scope.logical-expression.shell keyword.control.anchor.regexp.shell"
    },
    {
        "c": "a",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "(",
        "t": "source.shell source.shell meta.scope.logical-expression.shell meta.group.regexp.shell punctuation.definition.group.regexp.shell"
    },
    {
        "c": "[",
        "t": "source.shell source.shell meta.scope.logical-expression.shell meta.group.regexp.shell constant.other.character-class.set.regexp.shell punctuation.definition.character-class.regexp.shell"
    },
    {
        "c": "^",
        "t": "source.shell source.shell meta.scope.logical-expression.shell meta.group.regexp.shell constant.other.character-class.set.regexp.shell keyword.operator.negation.regexp.shell"
    },
    {
        "c": "a",
        "t": "source.shell source.shell meta.scope.logical-expression.shell meta.group.regexp.shell constant.other.character-class.set.regexp.shell"
    },
    {
        "c": "]",
        "t": "source.shell source.shell meta.scope.logical-expression.shell meta.group.regexp.shell constant.other.character-class.set.regexp.shell punctuation.definition.character-class.regexp.shell"
    },
    {
        "c": "+",
        "t": "source.shell source.shell meta.scope.logical-expression.shell meta.group.regexp.shell keyword.operator.quantifier.regexp.shell"
    },
    {
        "c": ")",
        "t": "source.shell source.shell meta.scope.logical-expression.shell meta.group.regexp.shell punctuation.definition.group.regexp.shell"
    },
    {
        "c": "a",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "(",
        "t": "source.shell source.shell meta.scope.logical-expression.shell meta.group.regexp.shell punctuation.definition.group.regexp.shell"
    },
    {
        "c": "[",
        "t": "source.shell source.shell meta.scope.logical-expression.shell meta.group.regexp.shell constant.other.character-class.set.regexp.shell punctuation.definition.character-class.regexp.shell"
    },
    {
        "c": "^",
        "t": "source.shell source.shell meta.scope.logical-expression.shell meta.group.regexp.shell constant.other.character-class.set.regexp.shell keyword.operator.negation.regexp.shell"
    },
    {
        "c": "a",
        "t": "source.shell source.shell meta.scope.logical-expression.shell meta.group.regexp.shell constant.other.character-class.set.regexp.shell"
    },
    {
        "c": "]",
        "t": "source.shell source.shell meta.scope.logical-expression.shell meta.group.regexp.shell constant.other.character-class.set.regexp.shell punctuation.definition.character-class.regexp.shell"
    },
    {
        "c": "+",
        "t": "source.shell source.shell meta.scope.logical-expression.shell meta.group.regexp.shell keyword.operator.quantifier.regexp.shell"
    },
    {
        "c": ")",
        "t": "source.shell source.shell meta.scope.logical-expression.shell meta.group.regexp.shell punctuation.definition.group.regexp.shell"
    },
    {
        "c": "a' ",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "]]",
        "t": "source.shell source.shell meta.scope.logical-expression.shell punctuation.definition.logical-expression.shell"
    },
    {
        "c": ";",
        "t": "source.shell source.shell punctuation.terminator.statement.semicolon.shell"
    },
    {
        "c": "[[",
        "t": "source.shell source.shell meta.scope.logical-expression.shell punctuation.definition.logical-expression.shell"
    },
    {
        "c": " ",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "$",
        "t": "source.shell source.shell meta.scope.logical-expression.shell punctuation.definition.variable.shell variable.other.normal.shell"
    },
    {
        "c": "var",
        "t": "source.shell source.shell meta.scope.logical-expression.shell variable.other.normal.shell"
    },
    {
        "c": " ",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "-r",
        "t": "source.shell source.shell meta.scope.logical-expression.shell keyword.operator.logical.shell"
    },
    {
        "c": "egex-match ",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "'",
        "t": "source.shell source.shell meta.scope.logical-expression.shell string.quoted.single.shell punctuation.definition.string.begin.shell"
    },
    {
        "c": "^a([^a]+)a([^a]+)a",
        "t": "source.shell source.shell meta.scope.logical-expression.shell string.quoted.single.shell"
    },
    {
        "c": "'",
        "t": "source.shell source.shell meta.scope.logical-expression.shell string.quoted.single.shell punctuation.definition.string.end.shell"
    },
    {
        "c": " ",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "]]",
        "t": "source.shell source.shell meta.scope.logical-expression.shell punctuation.definition.logical-expression.shell"
    },
    {
        "c": ";",
        "t": "source.shell source.shell punctuation.terminator.statement.semicolon.shell"
    },
    {
        "c": "[[",
        "t": "source.shell source.shell meta.scope.logical-expression.shell punctuation.definition.logical-expression.shell"
    },
    {
        "c": " ",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "$",
        "t": "source.shell source.shell meta.scope.logical-expression.shell punctuation.definition.variable.shell variable.other.normal.shell"
    },
    {
        "c": "var",
        "t": "source.shell source.shell meta.scope.logical-expression.shell variable.other.normal.shell"
    },
    {
        "c": " ",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "=~",
        "t": "source.shell source.shell meta.scope.logical-expression.shell keyword.operator.logical.shell"
    },
    {
        "c": "           '",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "^",
        "t": "source.shell source.shell meta.scope.logical-expression.shell keyword.control.anchor.regexp.shell"
    },
    {
        "c": "a",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "(",
        "t": "source.shell source.shell meta.scope.logical-expression.shell meta.group.regexp.shell punctuation.definition.group.regexp.shell"
    },
    {
        "c": "[",
        "t": "source.shell source.shell meta.scope.logical-expression.shell meta.group.regexp.shell constant.other.character-class.set.regexp.shell punctuation.definition.character-class.regexp.shell"
    },
    {
        "c": "^",
        "t": "source.shell source.shell meta.scope.logical-expression.shell meta.group.regexp.shell constant.other.character-class.set.regexp.shell keyword.operator.negation.regexp.shell"
    },
    {
        "c": "a",
        "t": "source.shell source.shell meta.scope.logical-expression.shell meta.group.regexp.shell constant.other.character-class.set.regexp.shell"
    },
    {
        "c": "]",
        "t": "source.shell source.shell meta.scope.logical-expression.shell meta.group.regexp.shell constant.other.character-class.set.regexp.shell punctuation.definition.character-class.regexp.shell"
    },
    {
        "c": "+",
        "t": "source.shell source.shell meta.scope.logical-expression.shell meta.group.regexp.shell keyword.operator.quantifier.regexp.shell"
    },
    {
        "c": ")",
        "t": "source.shell source.shell meta.scope.logical-expression.shell meta.group.regexp.shell punctuation.definition.group.regexp.shell"
    },
    {
        "c": "a",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "(",
        "t": "source.shell source.shell meta.scope.logical-expression.shell meta.group.regexp.shell punctuation.definition.group.regexp.shell"
    },
    {
        "c": "[",
        "t": "source.shell source.shell meta.scope.logical-expression.shell meta.group.regexp.shell constant.other.character-class.set.regexp.shell punctuation.definition.character-class.regexp.shell"
    },
    {
        "c": "^",
        "t": "source.shell source.shell meta.scope.logical-expression.shell meta.group.regexp.shell constant.other.character-class.set.regexp.shell keyword.operator.negation.regexp.shell"
    },
    {
        "c": "a",
        "t": "source.shell source.shell meta.scope.logical-expression.shell meta.group.regexp.shell constant.other.character-class.set.regexp.shell"
    },
    {
        "c": "]",
        "t": "source.shell source.shell meta.scope.logical-expression.shell meta.group.regexp.shell constant.other.character-class.set.regexp.shell punctuation.definition.character-class.regexp.shell"
    },
    {
        "c": "+",
        "t": "source.shell source.shell meta.scope.logical-expression.shell meta.group.regexp.shell keyword.operator.quantifier.regexp.shell"
    },
    {
        "c": ")",
        "t": "source.shell source.shell meta.scope.logical-expression.shell meta.group.regexp.shell punctuation.definition.group.regexp.shell"
    },
    {
        "c": "a' ",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "]]",
        "t": "source.shell source.shell meta.scope.logical-expression.shell punctuation.definition.logical-expression.shell"
    },
    {
        "c": ";",
        "t": "source.shell source.shell punctuation.terminator.statement.semicolon.shell"
    },
    {
        "c": "[[",
        "t": "source.shell source.shell meta.scope.logical-expression.shell punctuation.definition.logical-expression.shell"
    },
    {
        "c": " ",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "$",
        "t": "source.shell source.shell meta.scope.logical-expression.shell punctuation.definition.variable.shell variable.other.normal.shell"
    },
    {
        "c": "desc",
        "t": "source.shell source.shell meta.scope.logical-expression.shell variable.other.normal.shell"
    },
    {
        "c": " ",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "-r",
        "t": "source.shell source.shell meta.scope.logical-expression.shell keyword.operator.logical.shell"
    },
    {
        "c": "egex-match ",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "'",
        "t": "source.shell source.shell meta.scope.logical-expression.shell string.quoted.single.shell punctuation.definition.string.begin.shell"
    },
    {
        "c": "^alias for --(\\S+)",
        "t": "source.shell source.shell meta.scope.logical-expression.shell string.quoted.single.shell"
    },
    {
        "c": "'",
        "t": "source.shell source.shell meta.scope.logical-expression.shell string.quoted.single.shell punctuation.definition.string.end.shell"
    },
    {
        "c": " ",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "]]",
        "t": "source.shell source.shell meta.scope.logical-expression.shell punctuation.definition.logical-expression.shell"
    },
    {
        "c": ";",
        "t": "source.shell source.shell punctuation.terminator.statement.semicolon.shell"
    },
    {
        "c": "[[",
        "t": "source.shell source.shell meta.scope.logical-expression.shell punctuation.definition.logical-expression.shell"
    },
    {
        "c": " ",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "$",
        "t": "source.shell source.shell meta.scope.logical-expression.shell punctuation.definition.variable.shell variable.other.normal.shell"
    },
    {
        "c": "desc",
        "t": "source.shell source.shell meta.scope.logical-expression.shell variable.other.normal.shell"
    },
    {
        "c": " ",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "=~",
        "t": "source.shell source.shell meta.scope.logical-expression.shell keyword.operator.logical.shell"
    },
    {
        "c": "           '",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "^",
        "t": "source.shell source.shell meta.scope.logical-expression.shell keyword.control.anchor.regexp.shell"
    },
    {
        "c": "alias ",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "for",
        "t": "source.shell source.shell meta.scope.logical-expression.shell keyword.control.shell"
    },
    {
        "c": " --(",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "\\S",
        "t": "source.shell source.shell meta.scope.logical-expression.shell constant.character.escape.shell"
    },
    {
        "c": "+)",
        "t": "source.shell source.shell meta.scope.logical-expression.shell"
    },
    {
        "c": "'",
        "t": "source.shell source.shell meta.scope.logical-expression.shell string.quoted.single.shell punctuation.definition.string.begin.shell"
    },
    {
        "c": " ]];",
        "t": "source.shell source.shell meta.scope.logical-expression.shell string.quoted.single.shell"
    }
]
```
</details>

(Will send a pull request if I can fix)

**Comment by jeff-hykin** [comment]
Looks like the issue is here, caused by the logical expression pattern. I'm not sure when I'll be able to look into to in depth
```
    {
        ​"​c​"​: ​"​           '​"​,
        ​"​t​"​: ​"​source.shell source.shell meta.scope.logical-expression.shell​"​
    },
```

**Comment by jeff-hykin** [comment]
Should now be fixed on `v1.2.6`
